diff --git a/hw/vendor/openhwgroup_cv32e40p/rtl/cv32e40p_register_file_ff.sv b/hw/vendor/openhwgroup_cv32e40p/rtl/cv32e40p_register_file_ff.sv
index 4d31c12..07a7a07 100644
--- a/hw/vendor/openhwgroup_cv32e40p/rtl/cv32e40p_register_file_ff.sv
+++ b/hw/vendor/openhwgroup_cv32e40p/rtl/cv32e40p_register_file_ff.sv
@@ -66,7 +66,7 @@ module cv32e40p_register_file #(
   localparam NUM_WORDS = 2 ** (ADDR_WIDTH - 1);
   // number of floating point registers
   localparam NUM_FP_WORDS = 2 ** (ADDR_WIDTH - 1);
-  localparam NUM_TOT_WORDS = FPU ? (ZFINX ? NUM_WORDS : NUM_WORDS + NUM_FP_WORDS) : NUM_WORDS;
+  localparam NUM_TOT_WORDS = (FPU != 32'd0) ? ((ZFINX != 32'd0) ? NUM_WORDS : NUM_WORDS + NUM_FP_WORDS) : NUM_WORDS;
 
   // integer register file
   logic [    NUM_WORDS-1:0][DATA_WIDTH-1:0] mem;
