diff --git a/hw/vendor/openhwgroup_cv32e40p/rtl/cv32e40p_decoder.sv b/hw/vendor/openhwgroup_cv32e40p/rtl/cv32e40p_decoder.sv
index d03027b..8b16758 100644
--- a/hw/vendor/openhwgroup_cv32e40p/rtl/cv32e40p_decoder.sv
+++ b/hw/vendor/openhwgroup_cv32e40p/rtl/cv32e40p_decoder.sv
@@ -2923,7 +2923,7 @@ module cv32e40p_decoder
 
             // ZFINX
             CSR_ZFINX :
-                if (!COREV_PULP || (FPU && !ZFINX) || csr_op != CSR_OP_READ) begin
+                if ((COREV_PULP == 32'd0) || ((FPU != 32'd0) && (ZFINX != 32'd0)) || csr_op != CSR_OP_READ) begin
                   csr_illegal = 1'b1;
                 end
 
