diff --git a/hw/vendor/x-heep/hw/ip/pdm2pcm/rtl/pdm2pcm.sv.tpl b/hw/vendor/x-heep/hw/ip/pdm2pcm/rtl/pdm2pcm.sv.tpl
index 14da4fa..75d56eb 100644
--- a/hw/vendor/x-heep/hw/ip/pdm2pcm/rtl/pdm2pcm.sv.tpl
+++ b/hw/vendor/x-heep/hw/ip/pdm2pcm/rtl/pdm2pcm.sv.tpl
@@ -206,7 +206,7 @@ module pdm2pcm #(
 
   // Clock domain crossing FIFO
   cdc_fifo_gray #(
-      .T(logic [17:0]),
+      .T(logic [FIFO_WIDTH-1:0]),
       .LOG_DEPTH(FIFO_ADDR_WIDTH)
   ) pdm2pcm_fifo_i (
       .src_clk_i  (div_clk),
